src/api/posts.ts

```diff
@@ -0,0 +1,6 @@
+import { Post } from '../types/Post';
+import { client } from '../utils/fetchClient';
+
+export const getPosts = (userId: number) => {
+  return client.get<Post[]>(`/posts?userId=${userId}`);
+};
```

Comment:
Ensure that the `getPosts` function has an explicit return type definition to enhance type safety and prevent potential runtime errors.